### ISSUE TITLE
how to train？

### ISSUE THREAD

**Issue by fuzhao123232** [Jiyao06/GenPose#19]
Dear author, Thanks for your SOTA job. For train_score.sh, single card(4090), batch 192, when trained to 10 epochs, the loss decreased to 0.35-0.45, and then from 10-800 epochs, the train's loss kept jumping back and forth between 0.35-0.45, feeling it had converged. Then, evaluate and compare the author's paper. The average 5 ° 2cm loss was close to 10 points, and the values of other indicators were also low. That is to say, I reached a local optimum and couldn't decrease. How did the author train it?
![image](https://github.com/Jiyao06/GenPose/assets/115547529/6af1eb7a-bf97-4c28-97d3-bc8677c0e344)
my eval result:
![image](https://github.com/Jiyao06/GenPose/assets/115547529/fc712fe9-dc37-4756-ac89-c3ed5e71a337)
eval from author's kps:
![image](https://github.com/Jiyao06/GenPose/assets/115547529/eb288bb4-8985-48be-9caa-c473813c9eb9)

**Comment by fuzhao123232** [comment]
I try to train score network using 4 cards(A10 * 4), batch size = 192*4, and during epoch 172, the loss is as follow:
![image](https://github.com/Jiyao06/GenPose/assets/115547529/bfbdb07f-8e51-40c4-8e79-97b2801373c5)
during epoch 327
![image](https://github.com/Jiyao06/GenPose/assets/115547529/65690838-8faa-44c4-ad5e-59aac16cd467)
It seem to be converged,but the loss is bigger than sigle card(0.35).
I really don't know how to train the score network to reproduce the author's results.

**Comment by fuzhao123232** [comment]
The difference between my code and author's code is dataloader:
For to speed CPU IO, I save the pointclouds for get_ittem.
   
![image](https://github.com/Jiyao06/GenPose/assets/115547529/c4abceb4-291f-46e1-8be7-e6eb79bde285)


The pointcloud's generate script is as follow:
![image](https://github.com/Jiyao06/GenPose/assets/115547529/9381ef1b-35dd-43bc-aa30-e8d90c3edb1a)
![image](https://github.com/Jiyao06/GenPose/assets/115547529/65f49e90-1379-4d49-aae8-732b1d773556)
![image](https://github.com/Jiyao06/GenPose/assets/115547529/6be32e02-e4e2-4f45-bc94-6163d5e8a6c4)
![image](https://github.com/Jiyao06/GenPose/assets/115547529/8234ff4f-11a8-4b91-a5a2-19b4d2439fb9)


In fact, I have visual the results to check ,and I found the generated point cloud and RGB image correspond one-to-one.
![image](https://github.com/Jiyao06/GenPose/assets/115547529/b38f4ed2-b8e7-4837-b97a-6ed77de916e3)
![image](https://github.com/Jiyao06/GenPose/assets/115547529/13a5e828-f971-40db-a2bd-d3d2ea4f8880)
Therefore, I think my dataprocess is no problem. But the train is difficult.

**Comment by fuzhao123232** [comment]
![image](https://github.com/Jiyao06/GenPose/assets/115547529/829dd278-1e3f-44bb-8a45-188dff7f22d1)
请问作者论文中的结果是否使用了teacher model？最后的train loss大概降到了多少呢？

**Comment by Jiyao06** [comment]
1. To determine if the training process has converged, you can assess this by visualizing the training curve. 
2. Regarding your question about the higher loss when training with multiple GPUs, this could be due to the increased batch size without appropriately adjusting the learning rate. 
3. We did not use a teacher model during our training process.

**Comment by sseunghyuns** [comment]
Hello!
I wonder if you succeeded in reproducing the results of the paper.

**Comment by fuzhao123232** [comment]
[图片]
The results when I trained 320 epoch, Maybe when epoch reach to 1900, the results can be reproduced.

**Comment by fuzhao123232** [comment]
[图片]
The results when I trained 320 epoch, Maybe when epoch reach to 1900, the results can be reproduced. I need more time.
![image](https://github.com/Jiyao06/GenPose/assets/115547529/2a7ca8c9-6b61-4e80-bb89-3e3b701646fe)
안녕하세요, 친구, 이전 데이터 처리에 문제가 생겨서 일부 Real의 실제 데이터 훈련만 사용하여 결과를 재현할 수 없게 되었습니다.나는 지금 이 문제를 수정했다.그리고 이미 300여 개의 epoch를 훈련시켰다. 그림은 나의 평가 결과이고 loss는 여전히 떨어지고 있다. 그래서 나는 마지막으로 1900개의 epoch를 훈련해도 저자의 결과를 재현할 기회가 있다고 생각한다.

**Comment by sseunghyuns** [comment]
> [图片] The results when I trained 320 epoch, Maybe when epoch reach to 1900, the results can be reproduced. I need more time. ![image](https://private-user-images.githubusercontent.com/115547529/306840433-2a7ca8c9-6b61-4e80-bb89-3e3b701646fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2MTI1NDQsIm5iZiI6MTcwODYxMjI0NCwicGF0aCI6Ii8xMTU1NDc1MjkvMzA2ODQwNDMzLTJhN2NhOGM5LTZiNjEtNGU4MC1iYjg5LTNlM2I3MDE2NDZmZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMlQxNDMwNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNGVkMjhkNmY1ODZmOWEzYzkwYWRjZmJkOWZhNTNiODE4ZjBhMjQyYzQxNGJmMGRhN2M2NTI4OTdlMGNkZTdhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Z4vFrJtzxgo9OCqnZ8J_Vi7pF0fH901DQGzYGhQUPW4) 안녕하세요, 친구, 이전 데이터 처리에 문제가 생겨서 일부 Real의 실제 데이터 훈련만 사용하여 결과를 재현할 수 없게 되었습니다.나는 지금 이 문제를 수정했다.그리고 이미 300여 개의 epoch를 훈련시켰다. 그림은 나의 평가 결과이고 loss는 여전히 떨어지고 있다. 그래서 나는 마지막으로 1900개의 epoch를 훈련해도 저자의 결과를 재현할 기회가 있다고 생각한다.

Thanks!

**Comment by fuzhao123232** [comment]
eval when eopch is 1664:[图片]

**Comment by fuzhao123232** [comment]
eval when epoch is 1664:
![image](https://github.com/Jiyao06/GenPose/assets/115547529/c2c58c70-5de3-4670-87ac-c52b2a310a06)
compare to paper:
![image](https://github.com/Jiyao06/GenPose/assets/115547529/118bd737-a84b-461f-b580-b66e43ec95be)

I think maybe the train epoch will be > 2000, it really .....
Why is the convergence so slow? I will be cry.

**Comment by sseunghyuns** [comment]
Did you also trained EnergyNet? Or did you use author's pretrained checkpoint?
I'm also suffering slow convergence... Moreover I trained 1032 epochs and the evaluation score is much lower than yours @fuzhao123232.

<img width="251" alt="image" src="https://github.com/Jiyao06/GenPose/assets/63924704/57bd18b1-5fa6-4259-ad05-0a42b6c4c853">

**Comment by fuzhao123232** [comment]
I used author's pretrained checkpoint for EnergyNet and my Scorenet weights. I think the trained results might be relevent with the random initial weights。And I think you need to check your datasets is really ok？

**Comment by sseunghyuns** [comment]
Thank you! I think I have to check the dataloader. 
If it's fine, could you show me the loss curve recorded in the tensorboard?

**Comment by sseunghyuns** [comment]
> Thank you! I think I have to check the dataloader. If it's fine, could you show me the loss curve recorded in the tensorboard?

@fuzhao123232 It seems that my dataloader is fine. 

Also, in the comments above, when you trained 10 epochs, the loss dropped to 0.45. Was it always in the same boundary when you used single GPU or multiple GPUs? In my case, when I trained 100 epochs with single GPU(or 2 GPUs), the loss was 0.6-0.7. Did you make any changes to the default config settings?
(I'm referring to your code for pre-processing and loading the pcd data as npy files. When I visualized the dataloader, it seemed to be working fine.)

**Comment by sseunghyuns** [comment]
> > Thank you! I think I have to check the dataloader. If it's fine, could you show me the loss curve recorded in the tensorboard?
> 
> @fuzhao123232 It seems that my dataloader is fine.
> 
> Also, in the comments above, when you trained 10 epochs, the loss dropped to 0.45. Was it always in the same boundary when you used single GPU or multiple GPUs? In my case, when I trained 100 epochs with single GPU(or 2 GPUs), the loss was 0.6-0.7. Did you make any changes to the default config settings? (I'm referring to your code for pre-processing and loading the pcd data as npy files. When I visualized the dataloader, it seemed to be working fine.)

(24.03.14) There was a problem with ground truth label pkl. I think it's going to be solved if I handle this issue
